prow/plugins/mergecommitblocker/mergecommitblocker_test.go

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mergecommitblocker
+
+import (
+	"fmt"
+	"github.com/sirupsen/logrus"
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/github/fakegithub"
+	"reflect"
+	"testing"
+)
+const (
+	helpWanted = "help-wanted"
+	mergeCommit = "do-not-merge/contains-merge-commits"
+)
+type ghc struct {
+	*testing.T
+	labels    map[github.Label]bool
+	files     map[string][]byte
+	prChanges []github.PullRequestChange
+
+	addLabelErr, removeLabelErr, getIssueLabelsErr,
+	getFileErr, getPullRequestChangesErr error
+}
+
+func TestHandlePR(t *testing.T) {
+	var testcases = []struct {
+		name string
+		pullRequestEvent github.PullRequestEvent
+		commits          []github.RepositoryCommit
+		initialLabels  []string
+		addedLabel string
+		removedLabel   string
+	}{
+		{
+			name: "should label with do-not-merge/contains-merge-commits when merge commits present",
```

Comment:
I think you're missing cases for no-ops when the label is needed and exists, or is not needed and doesn't exist, or when GitHub gives errors. But these tests are sweet!

Reply:
I think we're still outstanding on those other test cases